_speedups.pyx

```diff
@@ -208,13 +229,19 @@ cdef class LocationStore:
                             yield entry.sender, entry.location, entry.item, entry.receiver, entry.flags
         elif slot_count:
             # generic implementation with lookup in set
+            receivers = ap_player_set_new(min(1000, slot_count))
```

Comment:
Why the min(1000 ?

Reply:
The 1000 could be any other number, but for 1000 slots you are very likely to have collisions (multiple items in the buckets) anyway. By limiting it to 1000 buckets, you limit the top level struct to ~8KB. The number is basically a trade-off between cache and cost of malloc, however we could preallocate bigger buckets if malloc was too slow.

Reply:
I added a comment to the code